src/main/java/org/jenkinsci/maven/plugins/hpi/RunMojo.java

```diff
@@ -443,6 +458,16 @@ public void configureWebApplication() throws Exception {
         
         super.configureWebApplication();
         getWebAppConfig().setWar(webAppFile.getCanonicalPath());
+        for (Artifact a : (Set<Artifact>) project.getArtifacts()) {
+            if (a.getGroupId().equals("org.jenkins-ci.main") && a.getArtifactId().equals("jenkins-core")) {
```

Comment:
This is only purporting to handle changes made in `jenkins-core`. Modules could perhaps be added later if there is demand.